django/db/backends/utils.py

```diff
@@ -125,9 +125,10 @@ def typecast_time(s):  # does NOT store time zone information
     hour, minutes, seconds = s.split(':')
     if '.' in seconds:  # check whether seconds have a fractional part
         seconds, microseconds = seconds.split('.')
+        microseconds = microseconds.ljust(6, '0')
```

Comment:
Yup it does seem to be a bit faster on both CPython 2 and 3

```py
In [1]: microseconds = '1'

In [3]: %timeit (microseconds + '000000')[:6]
The slowest run took 37.44 times longer than the fastest. This could mean that an intermediate result is being cached
10000000 loops, best of 3: 134 ns per loop

In [4]: %timeit '{:0<6.6}'.format(microseconds)
The slowest run took 11.50 times longer than the fastest. This could mean that an intermediate result is being cached
1000000 loops, best of 3: 352 ns per loop
```